cppcache/shared/CMakeLists.txt

```diff
@@ -47,6 +47,12 @@ set_target_properties(apache-geode PROPERTIES
   CXX_VISIBILITY_PRESET hidden
   VISIBILITY_INLINES_HIDDEN ON)
 
+if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU" AND ${USE_CPP_COVERAGE})
```

Comment:
Rather than duplicating these settings in each target how about using an `INTERFACE` target?

In root CMakeLists.txt:
```
add_library(INTERFACE _Coverage)
if(${CMAKE_CXX_COMPILER_ID} MATCHES "GNU" AND ${USE_CPP_COVERAGE})
  set_target_properties(_Coverage PROPERTIES
    COMPILE_FLAGS "--coverage"
    LINK_FLAGS "--coverage")
 endif()
```

In other CMakeLists:
```
target_link_libraries(<target>
  ...
  PRIVATE
  ...
  _Coverage)
```

The bonus is that if we add support for other compilers and linkers we only have to change the root CMakeLists.txt. The other will inherit the coverage flags from he `_Coverage` interface target.

Reply:
Was able to implement that and I like that it reduces duplicated CMake code.